bitcoin/src/crypto/key.rs

```diff
@@ -220,8 +217,8 @@ impl From<PublicKey> for XOnlyPublicKey {
 }
 
 /// An opaque return type for PublicKey::to_sort_key
-#[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
```

Comment:
`Copy` should stay, there's no reason to not have it. If `ArrayVec` doesn't support it we will have to figure out something else.

Reply:
Yes, `ArrayVec` does not support it for now. Should I close this PR then?

Reply:
So I investigated it and concluded the only solution is #2287.

If you rebase your PR on top of mine you should be able to simply remove `arrayvec` from `Cargo.toml` and use `internals::array_vec::ArrayVec` here and everything should work fine.

And for the future: the reason embedded failed is that you forgot to set `default-features = false` in `Cargo.toml`.

Reply:
Thanks for this. I'll just wait for it to get merged then I'll just rebase it on top of master